src/hierarchies/nnig_hierarchy.cpp

```diff
@@ -62,21 +74,25 @@ Eigen::VectorXd NNIGHierarchy::lpdf(const Eigen::MatrixXd &data) {
 //! \param data Column vector of data points
 //! \return     Marginal distribution vector evaluated in data
 Eigen::VectorXd NNIGHierarchy::eval_marg(const Eigen::MatrixXd &data) {
-  Eigen::VectorXd result = marg_lpdf(data);
+  Eigen::VectorXd result = marg_lpdf_grid(data);
   return result.array().exp();
 }
 
 //! \param data Column vector of data points
 //! \return     Marginal distribution vector evaluated in data (log)
-Eigen::VectorXd NNIGHierarchy::marg_lpdf(const Eigen::MatrixXd &data) {
-  // Get values of hyperparameters
-  double mu0 = get_mu0();
-  double lambda0 = get_lambda0();
-  double alpha0 = get_alpha0();
-  double beta0 = get_beta0();
+double NNIGHierarchy::marg_lpdf(const Eigen::MatrixXd &datum) {
+  assert(datum.rows() == 1 & datum.cols() == 1);
```

Comment:
Asserting cols==1 is not coherent with the hierarchy initializer function. I thought that I would not raise an error if cols>1, since it would be enough to take the first column (in previous iterations of the file a warning was printed acknowledging this). What's the better option?

Reply:
I'd rather always check that the data has correct form. I don't think it's wise being fault tolerant in this way, because the end user might feel frustrated (he is receiving a univariate output while he is possibly waiting for a multivariate one. He might have bugs in his code but would be looking in the wrong place)

Reply:
Fair enough. I've turned the warning in BaseAlgorithm into an error